Update API tests to deal with pagination and new options

kdelee opened the pull request:
Updates tests to allow the results of listing all credentials or sources to be
paginated. Also updates the equivlancy tests to set proper default values for
ssl_cert_verify and satellite version.

Closes #142

kdelee commented on camayoc/qcs_models.py:
```diff
@@ -330,9 +331,21 @@ def equivalent(self, other):
+                        sat_ver = self.options.get('satellite_version', '6.2')
+                    else:
+                        sat_ver = '6.2'
+                    if other.get(key).get('satellite_version') != sat_ver:
```
This logic is awkward and I'm interested to know if you have suggestion for some other flow.

elyezer replied:
I don't have any good suggestion right now. The only thing I see is that if `other` does not have `key` then it will return None and then will raise AttributeError since None object does not have the `get` method. I would recommend calling `other.get(key, {}).get('satellite_version')`.

kdelee commented on camayoc/tests/qcs/api/v1/sources/test_sources_common.py:
```diff
@@ -267,16 +267,18 @@ def test_read_all(src_type, cleanup, shared_client):
+    for server_source in server_srcs:
+        for created_source in created_srcs:
+            if server_source['id'] == created_source.fields()['id']:
+                assert_matches_server(created_source)
```
Now will fail on the guilty source

elyezer replied:
This is good but it will only catch the first "invalid" occurrence. What if multiple sources are "invalid"?

I am leaving this as something to think about. I mean we can go with this as the improvement for now, but as a future though I think we should start trying to provide all the problematic instances in the future.

elyezer commented on camayoc/tests/qcs/api/v1/sources/test_sources_common.py:
```diff
@@ -267,16 +267,18 @@ def test_read_all(src_type, cleanup, shared_client):
-                created_srcs.remove(cp)
+    server_srcs = Source().list().json()['results']
+    num_srcs = len(created_srcs)
+    for server_source in server_srcs:
```
I like having more descriptive names.

elyezer commented on camayoc/tests/qcs/api/v1/utils.py:
```diff
@@ -210,6 +210,8 @@ def prep_all_source_scan(cleanup, client, scan_type='inspect'):
             hosts=s['hosts'],
             credential_ids=[creds[s['credentials'][0]]],
         )
+        if s.get('options', False):
```
I am wondering if `src.options` will be defined by the constructor above? Even if assigned to `None`?

Because if it is set to `None` then this can become a one liner `src.options = s.get('options'[, default])` (define default if the default value is different from `None`

elyezer commented on tests/test_api.py:
```diff
@@ -59,16 +59,19 @@
-   'scan_type': 'connect',
-   'sources': [153],
-   'status': 'created'
+    'id': 21,
```
Four spaces :+1:

kdelee commented:
Fixed the linting problem and removed spurious formatting diff that the linter did not actually care about (spaces around + operator). I'm going to merge this, thanks for the review!

kdelee commented on camayoc/qcs_models.py:
```diff
@@ -330,9 +331,21 @@ def equivalent(self, other):
+                    else:
+                        ssl_verify = True
+                    if other.get(key, {}).get('ssl_cert_verify') != ssl_verify:
+                        return False
```
updated this RE: your comment

kdelee commented:
thanks!
